File: Speed Test/sum_to_n.cpp
// C++ code
#include <iostream>
#include <chrono>
#include <typeinfo>
#include "yaml-cpp/yaml.h"


long long sum_cpp(int n) {
    long long result = 0;
    for (int i = 1; i <= n; ++i) {
        result += i;
    }
    return result;
}

int main() {
    int n = 10000000;
    std::chrono::time_point<std::chrono::high_resolution_clock> start_time = std::chrono::high_resolution_clock::now();
//  std::cout << typeid(start_time).name() << std::endl;
    long long sum_result = sum_cpp(n);
    std::chrono::time_point<std::chrono::high_resolution_clock> end_time = std::chrono::high_resolution_clock::now(); // Wait for the user to respond.
    std::chrono::microseconds duration = std::chrono::duration_cast<std::chrono::microseconds>(end_time - start_time);
    std::cout << "C++: Sum of 1 to " << n << " is " << sum_result << " calculated in " << duration.count() / 1000000.0 << " seconds" << std::endl;
    return 0;
}

File: Speed Test/sum_to_n.py
#!/usr/bin/env python3

# Python code
import time
import yaml

def sum_python(n):
	result = 0
	for i in range(1, n + 1):
		result += i
	return result

n = 10000000
start_time = time.time()
sum_python(n)
end_time = time.time()
duration = start_time-end_time
print(f"Python: Sum of 1 to {n} is {sum_python(n)} calculated in {duration:.4f} seconds")
with open("Speed.yaml","r+") as f:
	metric = yaml.load(f, Loader= yaml.FullLoader)
	print(metric)
	metric["Sum from 1 to 10000000 Speed(seconds)"]["Python"] = duration
	
	yaml.dump(metric,stream=f,default_flow_style=False, sort_keys=False)
	print(metric)